src/shared/conf-parser.c

```diff
@@ -1185,3 +1185,35 @@ int config_parse_permille(const char* unit,
 
         return 0;
 }
+
+int config_parse_vlanprotocol(const char* unit,
+                              const char *filename,
+                              unsigned line,
+                              const char *section,
+                              unsigned section_line,
+                              const char *lvalue,
+                              int ltype,
+                              const char *rvalue,
+                              void *data,
+                              void *userdata) {
+        int *vlanprotocol = data;
+        assert(filename);
+        assert(lvalue);
+
+        if (isempty(rvalue)) {
+                *vlanprotocol = -1;
+                return 0;
+        }
+
+        if (STR_IN_SET(rvalue, "802.1ad", "802.1AD"))
+                *vlanprotocol = htobe16(ETH_P_8021AD);
+        else if (STR_IN_SET(rvalue, "802.1q", "802.1Q"))
+                *vlanprotocol = htobe16(ETH_P_8021Q);
```

Comment:
hmm, is there any particular reason why the BE conversion is done here during parsing, rather than when applying? we usually parse stuff into our structures in a resonably normalized form, and encoding endieness sounds unnormalized...

I mean byteswapping this here means whenever we access the field we have a good chance to forget it. But if we only convert only when passing it to the kernel we have much less of chance to fuck this up. Admittedly the field isn't used too often yet, but who knows if that will change one day?

At the very least, if we store non-native endianess fields in our structures we generally use types like "be32_t" or so, that indicate this fact, and so that "sparse" can tell us when we forget converting things. But quite frankly I wouldn't bother and just store this in native endian and only convert at the last moment.